src/rust/pal/linux/mod.rs

```diff
@@ -61,3 +62,15 @@ pub unsafe fn set_nonblock(fd: RawFd) -> i32 {
     flags |= libc::O_NONBLOCK;
     libc::fcntl(fd, libc::F_SETFL, flags, 1)
 }
+
+/// Converts a [std::net::SocketAddrV4] into a [libc::sockaddr_in].
+pub fn socketaddrv4_to_sockaddr_in(addr: &SocketAddrV4) -> libc::sockaddr_in {
```

Comment:
Ideally, this function wouldn't be here, it would be in src/rust/pal/sockets.rs (or somesuch), and it would return a value of a platform independent type, like pal::sockaddr_in (or somesuch).  And then there would be separate implementations of it for linux and Windows.  I.e. all platform-dependent code would be under pal, and everything in catcollor, catnap, catpowder, et cetera would be platform-independent.

Reply:
I agree that we should discuss a better approach for structuring the PAL module. Let us have it and then I handle the necessary reorg.